File: abc033/src/b.py
n = int(input())
sp = []
for i in range(n):
    s, p = input().split()
    sp.append((s, int(p)))

# loop through the list and add up all the populations
sum = sum(p for _, p in sp)

for s, p in sp:
    if p > sum // 2:
        print(s)
        break
else:
    print("atcoder")

File: abc033/src/bin/b.rs
use proconio::input;

fn main() {
    input! {
        n: usize,
        sp: [(String, i64); n],
    };
    // loop through the vector and add up all the populations
    let sum = sp.iter().map(|&(_, p_i)| p_i).sum::<i64>(); 

    for (s_i, p_i) in sp {
        if p_i > sum / 2 {
            println!("{}", s_i);
            return;
        }
    }
    println!("atcoder");
}